### DISCUSSION TITLE
feature request: method to suppress last login message

### DISCUSSION THREAD

**Discussion by koramangalacode** [ghostty-org/ghostty#3602] [Q&A] [answered]
Thanks for creating this incredible terminal.

I didn't see a discussion about this yet

> Last login: Fri Dec 27 03:33:11 on ttys001

This message is shown in every new tab and can be distracting. To disable it you can create a `.hushlogin` file in your home directory. But doing that suppresses the message when you ssh into your machine, when it actually is useful. Can a config option or method be added to disable it?

**Comment by koramangalacode** [comment]
## Potential solutions

1. Config option to hush login
`/usr/bin/login` will be run with the `-q` flag

2. Config option to override the `login` command
Terminal.app allows you to override the `login` command that initializes the shell

3. Building the app from source to add `-q` to the `login` command

4. Clearing the screen from your shell rc with a command like `echo $'\ec'`

5. Disabling or deleting Apple System Log (ASL) login files as root

6. A `HUSHLOGIN` environment variable that's configured in your macOS session and read by ghostty
 
`~/Library/LaunchAgents/environment.plist`
```xml
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">

<plist version="1.0">
<dict>
  <key>Label</key>
  <string>ghostty.environment</string>

  <key>ProgramArguments</key>
  <array>
    <string>sh</string>
    <string>-c</string>
    <string>launchctl setenv HUSHLOGIN 1</string>
  </array>

  <key>RunAtLoad</key>
  <true/>
</dict>
</plist>
```

**Reply by mitchellh** [reply]
Is this AI? This is... kind of sloppy. You absolutely cannot use `-q` with login for reasons noted in our source code. And that doesn't resolve the issue in the original post here that OTHER messages will also be noted.

**Comment by hssrrw** [comment]
Seems like creating the file `~/.hushlogin` is enough. Reading the source code comments, looks like `login` is already run with the `-q` flag for every window/tab regardless of the directory it was started in, if the .hushlogin file is present in the home directory.

https://github.com/ghostty-org/ghostty/blob/1fa2e699d142b9bc2795a4d3db98b035cc5a825c/src/termio/Exec.zig#L970-L975

**Reply by koramangalacode** [reply]
The reason why this is not the solution is in the discussion details

**Comment by mitchellh** [comment, marked as answer]
Ghostty doesn't print this message. This is a shell concern, or use hushlogin as noted (with its downsides also noted).

**Comment by rubnogueira** [comment]
Add `printf '\33c\e[3J'` to the beginning of your `.zshrc` file.

I believe that this is exclusive to macOS so you can do something like this:

```bash
[[ "$OSTYPE" == "darwin"* ]] && printf '\33c\e[3J'
```

Reference: https://stackoverflow.com/a/69915614
